### ISSUE TITLE
Some questions about the implementation details?

### ISSUE THREAD

**Issue by SparkJiao** [renll/ComerNet#1]
Very appreciate to your contribution!

I'm sorry to bother you but I have a few questions about the implementation details.

1. For all domains and slots, no matter how many tokens (subword level) each domain/slot is tokenized, the encoder always encodes it as a single vector representation,  which is also used as the weight for decoding, right? And another question is that have you ever tried use the hidden state of `[CLS]` as the whole embedding instead of the average pooling?

2. Indeed, I didn't figure out the exact meaning for the present `"The gradients are blocked for h1, h2, h3 during the backpropagation stage, since we only need them to work as the supplementary memories for the relation reasoning followed."` in the paper, since the operation of concatenation is nonparametric and all parameters except embedding weight is optimized during training [in the code](https://github.com/renll/ComerNet/blob/master/train.py#L106).

Thanks your very much!

**Comment by renll** [comment]
Thank you for your comment.

1. Yes, it is. Yes, I tried it and the accuracy dropped marginally (<0.5%) after the change.
2. When calculating the gradients for h1, h2, h3, we only consider the errors backpropagate from the residual connection and the attention module, and ignore the error source from the linear layer after the concatenation, since we blocked it. It is true that all the non-freezed paprameters are optimized, but the gradients applied to the parameters are different after gradient blocking.

By the way, the 'src_embedding' here is actually the freezed BERT encoding layer.
